src/core/router/info.h

```diff
@@ -51,67 +51,181 @@
   };
 
+  /// @return Char flag of given enumerated caps flag
+  /// @param flag Flag enum used for caps char flag
+  char GetTrait(CapsFlag flag) const noexcept
+  {
+    switch (flag)
+      {
+        case CapsFlag::Floodfill:
+          return 'f';
+
+        case CapsFlag::Hidden:
+          return 'H';
+
+        case CapsFlag::Reachable:
+          return 'R';
+
+        case CapsFlag::Unreachable:
+          return 'U';
+
+        case CapsFlag::LowBandwidth1:
+          return 'K';
+
+        case CapsFlag::LowBandwidth2:
+          return 'L';
+
+        case CapsFlag::HighBandwidth1:
+          return 'M';
+
+        case CapsFlag::HighBandwidth2:
+          return 'N';
+
+        case CapsFlag::HighBandwidth3:
+          return 'O';
+
+        case CapsFlag::HighBandwidth4:
+          return 'P';
+
+        case CapsFlag::UnlimitedBandwidth:
+          return 'X';
+
+        case CapsFlag::SSUTesting:
+          return 'B';
+
+        case CapsFlag::SSUIntroducer:
+          return 'C';
+
+        case CapsFlag::Unknown:
+        default:
+          return ' ';  // TODO(anonimal): review
+      }
+  }
+
+  /// @return Enumerated caps flag
+  /// @param value Char value of potential caps flag given
+  CapsFlag GetTrait(const char& value) const noexcept
+  {
+    if (value == GetTrait(CapsFlag::Floodfill))
+      return CapsFlag::Floodfill;
+
+    else if (value == GetTrait(CapsFlag::Hidden))
+      return CapsFlag::Hidden;
+
+    else if (value == GetTrait(CapsFlag::Reachable))
+      return CapsFlag::Reachable;
+
+    else if (value == GetTrait(CapsFlag::Unreachable))
+      return CapsFlag::Unreachable;
+
+    else if (value == GetTrait(CapsFlag::LowBandwidth1))
+      return CapsFlag::LowBandwidth1;
+
+    else if (value == GetTrait(CapsFlag::LowBandwidth2))
+      return CapsFlag::LowBandwidth2;
+
+    else if (value == GetTrait(CapsFlag::HighBandwidth1))
+      return CapsFlag::HighBandwidth1;
+
+    else if (value == GetTrait(CapsFlag::HighBandwidth2))
+      return CapsFlag::HighBandwidth2;
+
+    else if (value == GetTrait(CapsFlag::HighBandwidth3))
+      return CapsFlag::HighBandwidth3;
+
+    else if (value == GetTrait(CapsFlag::HighBandwidth4))
+      return CapsFlag::HighBandwidth4;
+
+    else if (value == GetTrait(CapsFlag::UnlimitedBandwidth))
+      return CapsFlag::UnlimitedBandwidth;
+
+    else if (value == GetTrait(CapsFlag::SSUTesting))
+      return CapsFlag::SSUTesting;
+
+    else if (value == GetTrait(CapsFlag::SSUIntroducer))
+      return CapsFlag::SSUIntroducer;
+
+    else
+      return CapsFlag::Unknown;  // TODO(anonimal): review
+  }
+
   struct Introducer {
     boost::asio::ip::address host;
-    int port;
+    std::uint16_t port{};
     Tag<32> key;
-    std::uint32_t tag;
-
-    /// @brief Human readable description of this struct
-    /// @param prefix for tabulations
-    /// @returns human readable string
-    std::string GetDescription(const std::string& tabs = std::string()) const;
```

Comment:
@MoroccanMalinois FYI, OOP-wise, I preferred your approach but, considering the use-case, the extra hoops we'd have to jump through to implement nested scope access to enclosed members would probably not have been in our best interest.